go/database/mpt/verification_proof.go

```diff
@@ -0,0 +1,336 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package mpt
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/Fantom-foundation/Carmen/go/common"
+	"github.com/Fantom-foundation/Carmen/go/common/amount"
+	"github.com/Fantom-foundation/Carmen/go/common/interrupt"
+	"github.com/Fantom-foundation/Carmen/go/common/witness"
+	"golang.org/x/exp/maps"
+	"math/rand"
+)
+
+// VerifyProofArchiveTrie verifies the consistency of witness proofs for an archive trie.
+// It reads the trie for each block within the input range.
+// It gets account and storage slots and extracts a witness proof for each account and its storage.
+// It is checked that values in the database and the proof match.
+func VerifyProofArchiveTrie(ctx context.Context, dir string, config MptConfig, from, to int, observer VerificationObserver) error {
+	trie, err := OpenArchiveTrie(dir, config, NodeCacheConfig{})
+	if err != nil {
+		return err
+	}
+
+	observer.StartVerification()
+	err = errors.Join(
+		verifyProofArchiveTrie(ctx, trie, from, to, observer),
+		trie.Close())
```

Comment:
nit: I think formatting it like this helps readability:
```suggestion
	err = errors.Join(
		verifyProofArchiveTrie(ctx, trie, from, to, observer),
		trie.Close(),
	)
```